src/main/java/it/gov/pagopa/payhub/activities/activity/treasury/TreasuryOpiIngestionActivityImpl.java

```diff
@@ -0,0 +1,80 @@
+package it.gov.pagopa.payhub.activities.activity.treasury;
+
+import it.gov.pagopa.payhub.activities.dao.IngestionFlowFileDao;
+import it.gov.pagopa.payhub.activities.dto.IngestionFlowFileDTO;
+import it.gov.pagopa.payhub.activities.dto.treasury.IufIuvDTO;
+import it.gov.pagopa.payhub.activities.dto.treasury.TreasuryIngestionResultDTO;
+import it.gov.pagopa.payhub.activities.exception.IngestionFlowFileNotFoundException;
+
+import it.gov.pagopa.payhub.activities.service.ingestionflow.IngestionFlowFileRetrieverService;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * Interface for the TreasuryOpiIngestionActivity.
+ * Defines methods for processing files based on an IngestionFlow ID.
+ */
+@Slf4j
+@Lazy
+@Component
+public class TreasuryOpiIngestionActivityImpl implements TreasuryOpiIngestionActivity {
+  private final String ingestionflowFileType;
+  private final IngestionFlowFileDao ingestionFlowFileDao;
+  private final IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService;
+
+
+  public TreasuryOpiIngestionActivityImpl(@Value("${ingestion-flow-file-type:O}") String ingestionflowFileType,
+                                          IngestionFlowFileDao ingestionFlowFileDao,
+                                          IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService) {
+    this.ingestionflowFileType = ingestionflowFileType;
+    this.ingestionFlowFileDao = ingestionFlowFileDao;
+    this.ingestionFlowFileRetrieverService = ingestionFlowFileRetrieverService;
+  }
+
+
+  @Override
+  public TreasuryIngestionResultDTO processFile(Long ingestionFlowFileId) {
+    List<IufIuvDTO> iufIuvList = new ArrayList<>();
+    try {
+      IngestionFlowFileDTO ingestionFlowFileDTO = findIngestionFlowFileRecord(ingestionFlowFileId);
+
+      List<Path> ingestionFlowFiles = retrieveFiles(ingestionFlowFileDTO);
+
+      ingestionFlowFiles.forEach(path -> {
+        File ingestionFlowFile = path.toFile();
+        log.debug("file from zip archive with name {} loaded successfully ", ingestionFlowFile.getName());
+
+      });
```

Comment:
don't create a File object just to read fileName, you could use Path#getFileName() instead.
Moreover, you are making a cycle just to print a debug log... if the actual level is not debug, you are looping for nothing (you could use log.isDebugEnabled to perform the action if needed).

Finally, for debug purporse, we could think to print the entire path, so at the very end it could be simplified as follow:
```suggestion
     log.debug("Successfully retrieved the following files related to the ingestionFlowFileId {}: {}", ingestionFlowFileId, ingestionFlowFiles);
```